server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/MailQueueName.java

```diff
@@ -0,0 +1,92 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+final class MailQueueName {
+
+    private static final String PREFIX = "JamesMailQueue";
+    private static final String EXCHANGE_PREFIX = PREFIX + "-exchange-";
+    @VisibleForTesting static final String WORKQUEUE_PREFIX = PREFIX + "-workqueue-";
+
+    static MailQueueName fromString(String name) {
+        Preconditions.checkNotNull(name);
+        return new MailQueueName(name);
+    }
+
+    static Optional<MailQueueName> fromRabbitWorkQueueName(String workQueueName) {
```

Comment:
I'm missing something in this class:
when are you getting a work queue name in James (ex.: _JamesMailQueue-workqueue-test_).

Reply:
when you list queues from Rabbit, you want to retrieve only queues that have the right prefix (that actually encodes the assumed type)

Reply:
I think the question of Antoine is the following: when you instantiate it with fromRabbitWorkQueueName, you can then call toRabbitExchangeName because you loose the type (work queue or exchange) you had first. Is this expected?

Reply:
Let's explain this class : 
MailQueueName is built with the James internal name (say "foo")
In Rabbit, I want to build artifacts (exchange, queue) that contains this name but are prefixed so we won't mess up easily with them.
The I created this class that, given the name ("foo") will be able to create the string JamesMailQueue-workqueue-test when I ask "what is the workqueue name for this MailQueueName ?" and JamesMailQueue-exchange-test when I ask "what is the exchange name for this MailQueueName ?".
Then, when I get a workqueue name from somewhere (likely rabbit management API) and I want to retrieve the James name, I use fromRabbitWorkQueueName. But the type of this object is always to same, I can just serialize this name for several purposes.
Does it make sense ?

Reply:
it does. It would have probably been more clear with different classes (one for workqueue, one for exchange)

Reply:
Yes agree, and MailQueueNa,e does a bunch of string handling that could be better understood with helper classes IMO

I contributed `WorkQueueName` and `ExchangeName` inner classes